### PR TITLE
Parameterless contructors for some CandleStickPatterns

### PR DESCRIPTION
**Pull request by Jay-Jay-D**
Parameterless contructors added to AbandonedBaby, DarkCloudCover, EveningDojiStar, EveningStar, MatHold, MorningDojiStar and MorningStar.

### CODE REVIEW

**Comment by mchandschuh** [comment]
It's strange to see five merge commits as part of this PR. Not sure what happened. Ideally the PR would only include the new commits (the last two)

**Comment by Jay-Jay-D** [comment]
[Maybe](https://xkcd.com/1597/) is because I made merge instead of rebase.

**Comment by mchandschuh** [comment]
Have you had a look at the [contributing.md](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md)? 

This document defines some good general practices for working with git. One key is that you always first refresh your origin and local master branches from QuantConnect, then perform changes on a new branch. Before submitting for a pull request, you can fetch and rebase your changes. This will help keep the source control history clean and understandable.

**Comment by Jay-Jay-D** [comment]
Sorry for the mess. 
Is there I can do to fix the problem now?

File: .gitignore [hunk 1]
```diff
@@ -251,4 +251,7 @@ Charts/
 
 # NCrunch files
 *.ncrunchsolution
-*.ncrunchproject
\ No newline at end of file
+*.ncrunchproject
+/Algorithm.CSharp/NT
+/Algorithm.CSharp/TrainingDataGenerator
+/Tests/TrainingDataGenerator
```
Comment (jaredbroad) [inline .gitignore]: Could you please revert the changes to this file?

File: UserInterface/QuantConnect.Views.csproj [hunk 2]
```diff
@@ -200,7 +200,9 @@
       <LastGenOutput>Resources.Designer.cs</LastGenOutput>
     </EmbeddedResource>
   </ItemGroup>
-  <ItemGroup />
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
+  </ItemGroup>
```
Comment (jaredbroad) [inline UserInterface/QuantConnect.Views.csproj]: Do you know what this is?

Comment (Jay-Jay-D) [inline UserInterface/QuantConnect.Views.csproj]: No, I don't know.

File: .gitignore (hunk 1 above)
Comment (Jay-Jay-D) [inline .gitignore]: Done.

**Comment by jaredbroad** [comment]
Nice work @Jay-Jay-D, thanks for seeing it through to the end :) I squashed all the intermediate merges together.

**Comment by Jay-Jay-D** [comment]
Thank you @jaredbroad @mchandschuh and @StefanoRaggi.
And again, sorry for the mess.

**Comment by jaredbroad** [comment]
No worries you should've seen some of my first commits on LEAN..  it read like this:
"Initial Commit"
"Initial Commit"
"Initial Commit"
"Initial Commit"
"Initial Commit"....
